ethsigner/core/src/main/java/tech/pegasys/ethsigner/core/requesthandler/sendtransaction/TransactionTransmitter.java

```diff
@@ -62,21 +63,25 @@ public TransactionTransmitter(
   }
 
   public void send() {
-    createSignedTransactionBody();
+    final Optional<JsonRpcRequest> request = createSignedTransactionBody();
+
+    if (request.isEmpty()) {
+      return;
+    }
+
+    try {
+      sendTransaction(Json.encodeToBuffer(request.get()));
+    } catch (final IllegalArgumentException e) {
+      LOG.debug("JSON Serialization failed for: {}", request, e);
+      routingContext.fail(BAD_REQUEST.code(), new JsonRpcException(INTERNAL_ERROR));
+    }
   }
 
-  private void createSignedTransactionBody() {
-    final String signedTransactionHexString;
+  private Optional<JsonRpcRequest> createSignedTransactionBody() {
     try {
```

Comment:
Perhaps split the updating of the nonce out into a seperate method

Reply:
yup - done